crates/common-domain/src/ids/macros.rs

```diff
@@ -0,0 +1,108 @@
+#[macro_export]
+macro_rules! id_type {
+    ($id_name:ident, $id_prefix:literal) => {
+        #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, Copy)]
+        #[cfg_attr(feature = "diesel", derive(diesel_derive_newtype::DieselNewType))]
+        #[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+        #[cfg_attr(feature = "utoipa", schema(value_type = String))]
+        pub struct $id_name(uuid::Uuid);
+
+        impl $id_name {
+            #[cfg(feature = "tonic")]
+            pub fn as_proto(&self) -> String {
+                self.to_string()
+            }
+
+            #[cfg(feature = "tonic")]
+            pub fn from_proto(value: &String) -> Result<$id_name, tonic::Status> {
+                $id_name::from_str(value.as_ref()).map_err(|_| {
+                    tonic::Status::invalid_argument(format!(
+                        "Invalid {}: {}",
+                        stringify!($id_name),
+                        value
+                    ))
+                })
+            }
```

Comment:
we have ProtoConv in our meteroid (main) crate, but we would need to implement it for every concrete id.
the bellow doesn't compile:
```rust
impl<ID> ProtoConv<String> for ID
    where
        ID: BaseId + FromStr + Display,
    {
        fn as_proto(&self) -> String {
            self.to_string()
        }

        fn from_proto_ref(proto: &String) -> Result<Self, tonic::Status> {
            ID::from_str(proto)
                .map_err(|_| tonic::Status::invalid_argument(format!("Invalid id: {}", proto)))
        }
    }
```
as in BaseId can be in future implemented for Datetime (that has already ProtoConv) so the compiler doesn't allow the code above to compile.